internal/map.go

```diff
@@ -55,3 +55,12 @@ func IfNotNil[From any, To any](value *From, mapper func(*From) To) To {
 	}
 	return *v
 }
+
+// ZeroIfNil returns the zero value if the variable is nil.
+func ZeroIfNil[T any](v *T) T {
+	if v == nil {
+		var zero T
+		return zero
+	}
+	return *v
+}
```

Comment:
s/zero/default value/

For non-integer types "zero" means something else, right? 

Do you expect this to be used for anything else than strings? If not, why not avoid the generic and make it a more readable variant of "get or get empty if nil" or more python-like "GetDefault"

Reply:
Answered here https://github.com/dapr-sandbox/components-go-sdk/pull/4#discussion_r974671584